ui/src/app/streams/stream-deploy/stream-deploy.service.ts

```diff
@@ -0,0 +1,245 @@
+      .pipe(
+        mergeMap(
+          val => this.streamsService.getDefinition(id),
+          (val1, val2) => {
+            return {
+              feature: val1,
+              streamDefinition: val2,
+            };
+          })
+      ).pipe(
+        mergeMap(
+          val => {
+            const observablesApplications = Parser.parse(val.streamDefinition.dslText as string, 'stream')
+              .lines[0].nodes
+              .map((node) => {
+                  return Observable.of({
+                    origin: node['name'],
+                    name: node['label'] || node['name'],
+                    type: node.type.toString(),
+                    version: null
+                  })
+                    .pipe(
+                      mergeMap(
+                        val1 => this.appsService.getAppVersions(ApplicationType[node.type], val1.origin),
+                        (val1: any, val2: AppVersion[]) => {
+                          val1.versions = val2;
+                          const current = val2.find((a: AppVersion) => a.defaultVersion);
+                          if (current) {
+                            val1.version = current.version;
+                          }
+                          return val1;
+                        }
+                      )
+                    );
+                }
+              );
+
+            if (val.feature.skipperEnabled) {
+              return Observable.forkJoin(this.streamsService.platforms(), ...observablesApplications);
+            }
+            return Observable.forkJoin(...observablesApplications);
+          },
+          (val, val2) => ({
+            feature: val.feature,
+            streamDefinition: val.streamDefinition,
+            args: val2,
+          }))
+      ).map(result => {
+        const config = new StreamDeployConfig();
+        config.skipper = result.feature.skipperEnabled;
+        config.id = id;
+
+        // Platform
+        if (config.skipper) {
+          const platforms = result.args[0] as Platform[];
+          (result.args as Array<any>).splice(0, 1);
+          config.platform = {
+            id: 'platform',
+            name: 'platform',
+            form: 'select',
+            type: 'java.lang.String',
+            defaultValue: '',
+            values: platforms.map((platform: Platform) => {
+              return {
+                key: platform.name,
+                name: platform.name,
+                type: platform.type
+              };
+            })
+          };
+
+        }
+        // Applications
+        config.apps = (result.args as Array<any>).map((app: any) => ({
+          origin: app.origin,
+          name: app.name,
+          type: app.type,
+          version: app.version,
+          versions: app.versions,
+          options: null,
+          optionsState: {
+            isLoading: false,
+            isOnError: false
+          }
+        }));
+
+        // Deployers
+        config.deployers = [
```

Comment:
Do we want to expose these deployer properties which are specific to the target platform (Local, CF and Kubernetes)? Though only the applicable properties would work on the target platform, not sure if we want to have a common list of these properties here.